redirect/src/main/java/com/adyen/checkout/redirect/RedirectUtil.kt

```diff
@@ -107,28 +84,98 @@ object RedirectUtil {
      * @return And intent that targets either another app or a Web page.
      */
     @JvmStatic
-    fun createRedirectIntent(context: Context, uri: Uri): Intent {
-        return if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R) {
-            createCustomTabIntent(context, uri)
-        } else {
-            if (determineResolveResult(context, uri).type == ResolveResult.Type.APPLICATION) {
-                Intent(Intent.ACTION_VIEW, uri)
-            } else {
-                createCustomTabIntent(context, uri)
+    fun launchUri(context: Context, uri: Uri) {
+        val launched = if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.R)
+            launchNativeApi30(context, uri)
+        else launchNativeBeforeApi30(context, uri)
+
+        if (!launched) {
+            val customTabsLaunch = launchWithCustomTabs(context, uri)
+            if (!customTabsLaunch) {
+                launchBrowser(context, uri)
             }
         }
     }
 
-    private fun createCustomTabIntent(context: Context, uri: Uri): Intent {
+    private fun launchNativeBeforeApi30(context: Context, uri: Uri): Boolean {
+        val pm = context.packageManager
+
+        // Get all Apps that resolve a generic url
+        val browserActivityIntent = Intent()
+            .setAction(Intent.ACTION_VIEW)
+            .addCategory(Intent.CATEGORY_BROWSABLE)
+            .setData(Uri.fromParts("http", "", null))
+        val genericPackageNameList: MutableList<String> =
+            pm.queryIntentActivities(browserActivityIntent, 0).map { it.resolvePackageName } as MutableList<String>
+        val genericResolvedList: Set<String> = genericPackageNameList.toSet()
+
+        // Get all apps that resolve the specific Url
+        val specializedActivityIntent = Intent(Intent.ACTION_VIEW, uri)
+            .addCategory(Intent.CATEGORY_BROWSABLE)
+        val specializedPackageNameList: MutableList<String> =
+            pm.queryIntentActivities(specializedActivityIntent, 0).map { it.resolvePackageName } as MutableList<String>
+        val resolvedSpecializedList: MutableSet<String> = specializedPackageNameList.toMutableSet()
```

Comment:
Can we use `val resolvedSpecializedList = pm.queryIntentActivities(specializedActivityIntent, 0).map { it.resolvePackageName }.toMutableList()` ?